src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2832,8 +2833,8 @@ protected RubyString reverse(RubyString string,
         }
 
         public static boolean reverseIsEqualToSelf(RubyString string,
-                CharacterLengthNode characterLengthNode) {
-            return characterLengthNode.execute(string.rope) <= 1;
+                TruffleString.CodePointLengthNode codePointLengthNode) {
+            return codePointLengthNode.execute(string.getTString(), string.encoding.tencoding) <= 1;
```

Comment:
Is this just a convention? Or does it make a difference in compiled code?

Reply:
No difference in compiled code, but it's basically the same convention as "Don't call `@Specialization` methods from outside the node", same thing here, "avoid calling `@ExportMessage` method outside the class".
For instance such a method might have a `@Cached` argument in the future.
Of course for this specific case it's unlikely and here it's mostly about consistency, all the other places use `.tstring`.
(There are some exceptions, like RubyString#isFrozen, oh well)

Reply:
Gotcha. I suspect it was picked because: a) it's considered best practice to call a getter method; and b) the IDE shows it as a completion option. We can mentally internalize the exception. I wonder if we could use modules to make the method's visibility narrower.

Reply:
>  it's considered best practice to call a getter method

FWIW I consider this overrated, because Java IDEs can easily refactor code to encapsulate a field if the getters/setters need to be more than just reading/writing the field.
So in new code I tend to prefer simply reading the field and making it public, to reduce noise, when the getters/setters would just access the field but do nothing else.